### DISCUSSION TITLE
BSH aktualisieren

### DISCUSSION THREAD

**Discussion by tolot27** [openjverein/jverein#356] [General]
Gibt es einen Grund, warum noch bsh-core-2.0b4 eingesetzt wird und keine aktuellere Version? Die alte Version unterstützt nämlich keine [Varargs](https://docs.oracle.com/javase/1.5.0/docs/guide/language/varargs.html). Diese wurden erst mit c1adf5d eingeführt.
Die Skripte von Lesefeldern, wo Methoden (z. B. `format`) verwendet werden, werden nicht richtig geparst und es kommen unsinnige Fehlermeldungen wie `Error in method invocation: Method format( java.lang.String, java.lang.Double ) not found in class'java.util.Formatter'`.

Ich weiß, es gibt kein offizielles Release mehr und nehme daher an, dass das der Grund ist. Vielleicht war es auch der Bug [arithmetic precedence bug](https://github.com/beanshell/beanshell/issues/590). Der ist jedoch auch schon länger behoben.
Das aktuelle Snapshot kann man von https://oss.sonatype.org/content/repositories/snapshots/org/beanshell/bsh/3.0.0-SNAPSHOT/ herunterladen.

**Comment by dippeal** [comment]
Kannst du gerne aktualisieren, testen und einen PR eröffnen.
